### PR TITLE
Operator Discovery + Kube 1.18

### PR DESCRIPTION
**Pull request by ecordell**
**Description of the change:**
Combination of #1425 and #1431 

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

**Comment by dinhxuanvu** [comment]
Waiting to update helm.

File: pkg/controller/operators/catalog/step_ensurer.go [hunk 1]
```diff
@@ -111,7 +111,7 @@ func (o *StepEnsurer) EnsureSecret(operatorNamespace, planNamespace, name string
 
 // EnsureBundleSecret creates user-specified secrets from the bundle. Called when StepResource.Secret is true
 func (o *StepEnsurer) EnsureBundleSecret(namespace string, secret *corev1.Secret) (status v1alpha1.StepStatus, err error) {
-	_, createErr := o.kubeClient.KubernetesInterface().CoreV1().Secrets(namespace).Create(secret)
+	_, createErr := o.kubeClient.KubernetesInterface().CoreV1().Secrets(namespace).Create(context.TODO(), secret, metav1.CreateOptions{})
```
Comment (awgreene) [inline pkg/controller/operators/catalog/step_ensurer.go:115]: Is this change a result of the bump to kube 1.18 in the next commit?

File: manifests/0000_50_olm_00-installplans.yaml [hunk 2]
```diff
@@ -1,304 +1,307 @@
...
+            type: object
+          spec:
+            description: InstallPlanSpec defines a set of Application resources to
+              be installed
+            type: object
+            required:
+            - approval
+            - approved
+            - clusterServiceVersionNames
+            - generation
```
Comment (awgreene) [inline manifests/0000_50_olm_00-installplans.yaml:62]: Should this be required?

File: deploy/chart/crds/0000_50_olm_00-catalogsources.yaml [hunk 3]
```diff
@@ -1,190 +1,193 @@
...
-  - name: v1alpha1
-    served: true
-    storage: true
-  preserveUnknownFields: false
-  scope: Namespaced
   names:
-    plural: catalogsources
-    singular: catalogsource
+    categories:
+    - olm
```
Comment (awgreene) [inline deploy/chart/crds/0000_50_olm_00-catalogsources.yaml:12]: What does `-olm` do in this case?

File: pkg/controller/operators/catalog/step_ensurer.go (hunk 1 above)
Comment (ecordell) [reply pkg/controller/operators/catalog/step_ensurer.go:115]: Yep!

File: manifests/0000_50_olm_00-installplans.yaml (hunk 2 above)
Comment (ecordell) [reply manifests/0000_50_olm_00-installplans.yaml:62]: Isn’t that fixed by the other PR you made?

File: deploy/chart/crds/0000_50_olm_00-catalogsources.yaml (hunk 3 above)
Comment (ecordell) [reply deploy/chart/crds/0000_50_olm_00-catalogsources.yaml:12]: You can “kubectl get olm” to get all olm related resources

Comment (awgreene) [reply deploy/chart/crds/0000_50_olm_00-catalogsources.yaml:12]: Great!

**Comment by ecordell** [comment]
/hold 

waiting on https://github.com/openshift/cluster-version-operator/pull/352

**Comment by njhale** [comment]
I re-rebased over in #1468

**Pull request closed without merging**